pkg/services/sqlstore/migrations/accesscontrol/migrations.go

```diff
@@ -170,4 +170,20 @@ func AddMigration(mg *migrator.Migrator) {
 	mg.AddMigration("add column hidden to role table", migrator.NewAddColumnMigration(roleV1, &migrator.Column{
 		Name: "hidden", Type: migrator.DB_Bool, Nullable: false, Default: "0",
 	}))
+
+	mg.AddMigration("permission kind migration", migrator.NewAddColumnMigration(permissionV1, &migrator.Column{
```

Comment:
This is a difficult one - I think in general we try to avoid putting migrations behind feature toggles. But if we decide to not go ahead with this approach and remove these migrations, it would be a breaking change. So you're right that adding a feature toggle here is safer. I'll go ahead with your suggestion and add a feature toggle to protect this.

Reply:
I just remembered why I didn't put these migrations behind a feature toggle - it would mean that we can't add the new fields to the [permission struct](https://github.com/grafana/grafana/pull/71933/files#diff-a3d308124dd62f2cf2d8fb23a28e2fc90d7319b81b93c1960196a074d9bcf310L179-L187), as we use this struct to query the permission table and these columns won't exist in the table. I think running these behind a feature toggle would be pretty difficult (we would need two different structs for permissions), and the risk probably isn't worth the effort. What do you think?

I will undo the change of adding the feature toggle now, and will also remove the uniqueness constraint (because there will be a lot of rows with empty kind, attribute and identifier for instances that have the feature toggle disabled).

Reply:
Arf! I had overlooked the xorm magic, yes it's be a pain to update every sql request :+1:
Your points make sense, thanks :100: